### PR TITLE
SN-441 add breadcrumbs to entity details page

### PR DESCRIPTION
**Pull request by toanphamsaritasa**
https://saritasa.atlassian.net/browse/SN-441

### CODE REVIEW

**Comment by toanphamsaritasa** [comment]
When go to entity detail page, there is a breadcrumbs navigation
image
Clicking on "Entities" will redirect to entities page.
image
Clicking on "Home" will redirect to homepage of web application.

**Review by AlexanderMakarov124 — APPROVED** [review]

File: src/Saritasa.NetForge.Blazor/Pages/EntityDetails.razor.cs [hunk 1]
```diff
@@ -15,9 +16,27 @@ public partial class EntityDetails : MvvmComponentBase<EntityDetailsViewModel>
     [Parameter]
     public string StringId { get; set; } = null!;
 
+    private EntityDetailsViewModel entityDetailsViewModel = null!;
+
+    private readonly List<BreadcrumbItem> breadcrumbItems = new()
+    {
+        new BreadcrumbItem("Home", href: "/"),
+        new BreadcrumbItem("Entities", href: "/admin"),
```
Comment (vasiliy-chefonov) [inline src/Saritasa.NetForge.Blazor/Pages/EntityDetails.razor.cs:24]: It shouldn't be hardcoded. Users can actually set the custom value for that using the `UseEndpoint` from the `AdminOptionsBuilder`.

File: src/Saritasa.NetForge.Blazor/Pages/EntityDetails.razor.cs [hunk 2]
```diff
@@ -15,9 +16,27 @@ public partial class EntityDetails : MvvmComponentBase<EntityDetailsViewModel>
     [Parameter]
     public string StringId { get; set; } = null!;
 
+    private EntityDetailsViewModel entityDetailsViewModel = null!;
+
+    private readonly List<BreadcrumbItem> breadcrumbItems = new()
+    {
+        new BreadcrumbItem("Home", href: "/"),
```
Comment (vasiliy-chefonov) [inline src/Saritasa.NetForge.Blazor/Pages/EntityDetails.razor.cs:23]: Let's leave only the Entities, no Home. View Site link is enough for the main page navigation I guess.

File: src/Saritasa.NetForge.Blazor/Pages/EntityDetails.razor.cs [hunk 3]
```diff
@@ -15,9 +16,27 @@ public partial class EntityDetails : MvvmComponentBase<EntityDetailsViewModel>
...
     /// <inheritdoc/>
     protected override EntityDetailsViewModel CreateViewModel()
     {
-        return ViewModelFactory.Create<EntityDetailsViewModel>(StringId);
+        entityDetailsViewModel = ViewModelFactory.Create<EntityDetailsViewModel>(StringId);
+        return entityDetailsViewModel;
+    }
+
+    /// <inheritdoc />
+    protected override void OnParametersSet()
```
Comment (vasiliy-chefonov) [inline src/Saritasa.NetForge.Blazor/Pages/EntityDetails.razor.cs:35]: Can we override `OnInitializedComponentAsync`?

File: src/Saritasa.NetForge.Blazor/Pages/EntityDetails.razor.cs [hunk 4]
```diff
@@ -15,9 +16,27 @@ public partial class EntityDetails : MvvmComponentBase<EntityDetailsViewModel>
...
+    {
+        new BreadcrumbItem("Home", href: "/"),
+        new BreadcrumbItem("Entities", href: "/admin"),
+    };
+
     /// <inheritdoc/>
     protected override EntityDetailsViewModel CreateViewModel()
     {
-        return ViewModelFactory.Create<EntityDetailsViewModel>(StringId);
+        entityDetailsViewModel = ViewModelFactory.Create<EntityDetailsViewModel>(StringId);
```
Comment (vasiliy-chefonov) [inline src/Saritasa.NetForge.Blazor/Pages/EntityDetails.razor.cs:30]: what is the purpose of this change? Can we just use `this.ViewModel`?

**Review by vasiliy-chefonov — CHANGES_REQUESTED** [review]

File: src/Saritasa.NetForge.Blazor/Pages/EntityDetails.razor.cs (hunk 3 above)
Comment (toanphamsaritasa) [reply src/Saritasa.NetForge.Blazor/Pages/EntityDetails.razor.cs:35]: From the document of [ComponentBase.OnInitializedAsync](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.components.componentbase.oninitializedasync?view=aspnetcore-7.0) indicates that

> Method invoked when the component is ready to start, having received its initial parameters from its parent in the render tree.

While in this method, we need to get the parameter and put it into the list for the breadcrumbs before initialize the ` MudBreadcrumbs` component. Therefore, I think it's better to use `OnParametersSet`. Also, I tried to replace it with `OnInitializedComponentAsync` and the breadcrumbs navigation is not working properly.

**Review by vasiliy-chefonov — APPROVED** [review]

**Pull request merged**